docs/api.rst

```diff
@@ -1,23 +1,203 @@
+   dictionary of supported `object-description-options`.
+
+   This option may be used to specify options for a single option type.  To
+   specify options for multiple object types at once, use
+   :confval:`object_description_options_patterns` instead.
+
+.. confval:: object_description_options_patterns
+
+   :python:`list` of :python:`(pattern, options)` pairs, where :python:`pattern`
+   is a regular expression matching strings of the form
+   :python:`"domain:objtype"` and :python:`options` is a dictionary of supported
+   `object-description-options`.
+
+The actual options for a given object type are determined as follows:
+
+1. The options are initialized to the defaults indicated for each of the
+   `object-description-options`.
+
+2. Any options specified for the exact object type by
+   :confval:`object_description_options_map` are applied as overrides.
+
+3. For each matching pattern in :confval:`object_description_options_pattern` in
+   order, the specified options are applied as additional overrides.
+
+.. _object-description-options:
+
+Object description options
+--------------------------
+
+The following options can be customized for each object type using either
+:confval:`object_description_options_map` or
+:confval:`object_description_options_patterns`.
+
+.. objconf:: include_in_toc
+
+   `bool` that indicates whether to include the object description in the table
+   of contents.  Defaults to :python:`True`.
+
+   .. admonition:: Example
+      :class: example
+
+      To prevent C++ parameter descriptions from appearing in the TOC, add the
+      following to :file:`conf.py`:
+
+      .. code-block:: python
+
+         object_description_options_paterns = [
+             ("cpp:.*Param": dict(include_in_toc=False)),
+         ]
+
+.. objconf:: generate_synopses
+
+   Indicates whether to generate a *synopsis* from the object description.  The
+   synopsis is shown as a tooltip when hovering over a cross-reference link, and
+   is also shown in the search results list.  Supported values are:
+
+   :python:`None`
+     Disables synopsis generation.
+
+   :python:`"first_paragraph"`
+     Uses the first paragraph of the description as the synopsis.
+
+   :python:`"first_sentence"`
+     Uses the first sentence of the first paragraph of the description as the synopsis.
+
+   The default is :python:`"first_paragraph"` except for :regexp:`c(pp)?:.*Param`
+   where the default is :python:`"first_sentence"`.
+
+   .. note::
+
+      Synopsis generation is currently supported only for the following domains:
+
+      - ``std`` (including object types added using :py:obj:`sphinx.application.Sphinx.add_object_type`)
+      - ``c`` and ``cpp``
+
+   .. admonition:: Example
+      :class: example
+
+      To use the first sentence rather than the first paragraph as the synopsis
+      for C++ class descriptions, add the following to :file:`conf.py`:
+
+      .. code-block:: python
+
+         object_description_options_map = [
```

Comment:
Should be a dict or ``object_description_options_patterns`

Reply:
Fixed.